dotCMS/src/main/java/com/dotmarketing/servlets/ShortyServlet.java

```diff
@@ -508,35 +521,72 @@ protected final String inodePath(final Contentlet contentlet,
         final Optional<Field> fieldOpt = resolveField(contentlet, tryField);
 
         if (fieldOpt.isEmpty()) {
-            return "/" + contentlet.getInode() + "/" + FILE_ASSET_DEFAULT;
+            return buildFieldPath(contentlet, FILE_ASSET_DEFAULT);
         }
 
         final Field field = fieldOpt.get();
         if (field instanceof ImageField || field instanceof FileField) {
 
             final String relatedImageId = contentlet.getStringProperty(field.variable());
-            final Optional<ContentletVersionInfo> contentletVersionInfo =
+            Optional<ContentletVersionInfo> contentletVersionInfo =
                     this.versionableAPI.getContentletVersionInfo(relatedImageId, contentlet.getLanguageId());
 
-            if (contentletVersionInfo.isPresent()) {
-                final String inode = live ? contentletVersionInfo.get().getLiveInode()
-                        : contentletVersionInfo.get().getWorkingInode();
+            if (contentletVersionInfo.isEmpty() && shouldFallbackToDefaultLanguage(contentlet)) {
+                // Try finding the contentlet version with the default language ID
+                contentletVersionInfo = this.versionableAPI.getContentletVersionInfo(relatedImageId,APILocator.getLanguageAPI().getDefaultLanguage().getId());
+            }
 
-                final Contentlet imageContentlet = APILocator.getContentletAPI()
-                        .find(inode, APILocator.systemUser(), false);
+            if (contentletVersionInfo.isPresent()) {
+                final Contentlet imageContentlet = getImageContentlet(contentletVersionInfo.get(), live);
+                validateContentlet(imageContentlet, live, imageContentlet.getInode());
+                final String fieldVar = imageContentlet.isDotAsset() ? DotAssetContentType.ASSET_FIELD_VAR : FILE_ASSET_DEFAULT;
+                return buildFieldPath(imageContentlet, fieldVar);
+            }
 
-                validateContentlet(imageContentlet, live, inode);
+        }
+        return buildFieldPath(contentlet, field.variable());
+    }
 
-                final String fieldVar = imageContentlet.isDotAsset() ?
-                        DotAssetContentType.ASSET_FIELD_VAR : FILE_ASSET_DEFAULT;
+    /**
+     * Determines whether the system should attempt to fallback to the default language
+     * for the given contentlet. This is used when content is not found in the
+     * contentlet's original language.
+     *
+     * @param contentlet The contentlet to check for language fallback eligibility
+     * @return true if the system should attempt to use the default language, false otherwise
+     */
+    private boolean shouldFallbackToDefaultLanguage(final Contentlet contentlet) {
+        final long defaultLanguageId = APILocator.getLanguageAPI().getDefaultLanguage().getId();
+        return defaultLanguageId != contentlet.getLanguageId() &&
+                APILocator.getLanguageAPI().canDefaultFileToDefaultLanguage();
```

Comment:
I think we can re-order this condition so if APILocator.getLanguageAPI().canDefaultFileToDefaultLanguage() isn't true we don't need to perform the other operations is just a matter of moving it up.